Source/ui_unix/vfsmodel.cpp

```diff
@@ -0,0 +1,96 @@
+#include "vfsmodel.h"
+#include "vfsmanagerdialog.h"
+
+VFSModel::VFSModel(QObject *parent)
+    :QAbstractTableModel(parent)
+{
+    SetupDevices();
+}
+
+int VFSModel::rowCount(const QModelIndex & /*parent*/) const
+{
+   return m_devices.size();
+}
+
+int VFSModel::columnCount(const QModelIndex & /*parent*/) const
+{
+    if (m_devices.size() > 0) {
+        return 3;
+    } else {
+        return 0;
+    }
+}
+
+QVariant VFSModel::data(const QModelIndex &index, int role) const
+{
+    if (role == Qt::DisplayRole)
+    {
+        CDevice* m_device = m_devices.at(index.row());
+        const char* val;
```

Comment:
This should be initialized (probably to an empty string). Just trying to avoid UB.